module4/src/controllers/users.js

```diff
@@ -0,0 +1,104 @@
+const express = require('express');
+const httpStatusCodes = require('../config/httpStatusCodes');
+const User = require('../models/user');
+const { createUser, deleteUser, getUser, getUsers, updateUser } = require('../services/users');
+const { addUsersToGroup, deleteRecord } = require('../services/userGroup');
+
+const api = express.Router();
+
+api.get('/list', async (req, res) => {
+    const { limit = 10, filter = '' } = req.query;
+
+    const result = await getUsers(limit, filter);
+
+    if (result.error) {
+        const { message } = result.error;
+        res.status(httpStatusCodes.BAD_REQUEST).send(message);
+    } else {
+        const { data } = result;
+        res.status(httpStatusCodes.OK).send(data);
+    }
+});
+
+api.get('/getUserById', async (req, res) => {
+    const { id: userId } = req.query;
+
+    const result = await getUser(userId);
+
+    if (result.error) {
+        const { message } = result.error;
+        res.status(httpStatusCodes.BAD_REQUEST).send(message);
+    } else {
+        const { data, message } = result;
+        if (data) {
```

Comment:
Can be replaced with the following code below to avoid code duplication.

```
const sendData = data || message;

res.status(httpStatusCodes.OK).send(sendData)
```